docs: Update migration-guide.md (#2021)
GoogleCloudPlatform/cloud-sql-proxy: migration-guide.md

./cloud-sql-proxy --address 0.0.0.0 --port 6000 <INSTANCE_CONNECTION_NAME>
```

## Environment variable changes

In v1 it was possible to do this:

``` shell
export INSTANCES="<INSTANCE_CONNECTION_NAME_1>=tcp:3307,<INSTANCE_CONNECTION_NAME_2>=tcp:5432"

./cloud_sql_proxy
```

In v2, we've significantly expanded the support for environment variables.
All flags can be set with an environment variable including instance connection names.

For example, in v2 this is possible:

```
export CSQL_PROXY_INSTANCE_CONNECTION_NAME_0="<INSTANCE_CONNECTION_NAME_1>?port=3307"

> export CSQL_PROXY_INSTANCE_CONNECTION_NAME_1="<INSTANCE_CONNECTION_NAME_2>?port=5342"

Reviewer: ```suggestion
export CSQL_PROXY_INSTANCE_CONNECTION_NAME_1="<INSTANCE_CONNECTION_NAME_2>?port=5432"
```

./cloud-sql-proxy --address 0.0.0.0 --port 6000 <INSTANCE_CONNECTION_NAME>
```

## Environment variable changes

In v1 it was possible to do this:

``` shell

> export INSTANCES="<INSTANCE_CONNECTION_NAME_1>,<INSTANCE_CONNECTION_NAME_2>"

Reviewer: does this need the `=tcp...` at the end of the instance connection names?

./cloud_sql_proxy
```

In v2, we've significantly expanded the support for environment variables.

> All flags can be set with an environment variable. And instance connection names
> can be set with variables.

Reviewer: ```suggestion
All flags can be set with an environment variable, this includes setting
instance connection names.
```
